packages/turf-center-median/index.js

```diff
@@ -0,0 +1,124 @@
+ *
+ * @name centerMedian
+ * @param {FeatureCollection<Point>} features Feature Collection of GeoJSON Points
+ * @param {Object} [options={}] Optional parameters
+ * @param {string} [options.weight] the property name used to weight the center
+ * @param {number} [options.tolerance=0.001] the difference in distance between candidate medians at which point the algorighim stops iterating.
+ * @param {number} [options.counter=10] how many attempts to find the median, should the tolerance be insufficient.
+ * @returns {Feature<Point>} The median center of the collection
+ * @example
+ * var points = turf.points([[0, 0], [1, 0], [0, 1], [5, 8]]);
+ * var medianCenter = turf.centerMedian(points);
+ *
+ * //addToMap
+ * var addToMap = [points, medianCenter]
+ */
+function centerMedian(features, options) {
+    // Optional params
+    options = options || {};
+    if (!isObject(options)) throw new Error('options is invalid');
+    var counter = options.counter || 10;
+    if (!isNumber(counter)) throw new Error('counter must be a number');
+    var weightTerm = options.weight;
+
+    // Calculate mean center:
+    var meanCenter = centerMean(features, {weight: options.weight});
+
+    // Calculate center of every feature:
+    var centroids = featureCollection([]);
+    featureEach(features, function (feature) {
+        centroids.features.push(centroid(feature, {weight: feature.properties[weightTerm]}));
+    });
+
+    centroids.properties = {
+        tolerance: options.tolerance,
+        medianCandidates: []
+    };
+    return findMedian(meanCenter, [0, 0], centroids, counter);
+}
+
+/**
+ * Recursive function to find new candidate medians.
+ *
+ * @private
+ * @param {Feature<Point>} candidateMedian current candidate median
+ * @param {Feature<Point>} previousCandidate the previous candidate median
+ * @param {FeatureCollection<Point>} centroids the collection of centroids whose median we are determining
+ * @param {number} counter how many attempts to try before quitting.
+ * @returns {Feature<Point>} the median center of the dataset.
+ */
+function findMedian(candidateMedian, previousCandidate, centroids, counter) {
+    var tolerance = centroids.properties.tolerance || 0.001;
+    var candidateXsum = 0;
+    var candidateYsum = 0;
+    var kSum = 0;
+    var centroidCount = 0;
+    featureEach(centroids, function (theCentroid) {
+        var weightValue = theCentroid.properties.weight;
+        var weight = (weightValue === undefined || weightValue === null) ? 1 : weightValue;
+        weight = Number(weight);
+        if (!isNumber(weight)) throw new Error('weight value must be a number');
+        if (weight > 0) {
+            centroidCount += 1;
+            var distanceFromCandidate = weight * distance(theCentroid, candidateMedian);
+            if (distanceFromCandidate === 0) distanceFromCandidate = 1;
+            var k = weight / distanceFromCandidate;
+            candidateXsum += getCoord(theCentroid)[0] * k;
```

Comment:
@muziejus Using `getCoord` does "look" good, however it does do some unnecessary processing.

This method is mostly used to clean the input data, you don't need to use `getCoord` if you are handling the output of a TurfJS method since the output will always be the same type.

Since you are iterating over a FeatureCollection of Point using `featureEach`, then `theCentroid` will ALWAYS be a `Feature<any>` which means if we access `theCentroid.geometry.coordinates` it will always be present without throwing any errors.

Everytime `getCoord` is fired, the following needs to be executed (which slows down the overall performance).

https://github.com/Turfjs/turf/blob/master/packages/turf-invariant/index.js#L15-L26

```js
export function getCoord(obj) {
    if (!obj) throw new Error('obj is required');

    var coordinates = getCoords(obj);

    // getCoord() must contain at least two numbers (Point)
    if (coordinates.length > 1 && isNumber(coordinates[0]) && isNumber(coordinates[1])) {
        return coordinates;
    } else {
        throw new Error('Coordinate is not a valid Point');
    }
}
```

Reply:
OK thanks.